### ISSUE TITLE
`vector-builder-0.3.4` fails on older GHC versions

### ISSUE THREAD

**Issue by Gabriella439** [nikita-volkov/vector-builder#2]
```
library/VectorBuilder/MonadPlus.hs:17:3:

    Could not deduce (Functor m) arising from a use of ‘fmap’

    from the context (MonadPlus m)

      bound by the type signature for

                 VectorBuilder.MonadPlus.many :: MonadPlus m => m a -> m (Vector a)

      at library/VectorBuilder/MonadPlus.hs:15:9-42

    Possible fix:

      add (Functor m) to the context of

        the type signature for

          VectorBuilder.MonadPlus.many :: MonadPlus m => m a -> m (Vector a)

    In the expression: fmap B.build loop

    In an equation for ‘many’:

        many m

          = fmap B.build loop

          where

              loop

                = mplus

                    (do { !element <- m;

                          remainders <- loop;

                          .... })

                    (return mempty)

library/VectorBuilder/MonadPlus.hs:35:3:

    Could not deduce (Functor m) arising from a use of ‘fmap’

    from the context (MonadPlus m)

      bound by the type signature for

                 sepBy1 :: MonadPlus m =>

                           m element -> m separator -> m (Vector element)

      at library/VectorBuilder/MonadPlus.hs:33:11-71

    Possible fix:

      add (Functor m) to the context of

        the type signature for

          sepBy1 :: MonadPlus m =>

                    m element -> m separator -> m (Vector element)

    In the expression: fmap B.build loop

    In an equation for ‘sepBy1’:

        sepBy1 elementM separatorM

          = fmap B.build loop

          where

              loop

                = do { !element <- elementM;

                       .... }

Failed to install vector-builder-0.3.4
```

**Comment by nikita-volkov** [comment]
Should be fixed in 0.3.4.1

**Comment by Gabriella439** [comment]
Thanks!
